src/Microsoft.IdentityModel.Tokens/BaseConfigurationManager.cs

```diff
@@ -35,9 +41,27 @@ public TimeSpan AutomaticRefreshInterval
                     throw LogHelper.LogExceptionMessage(new ArgumentOutOfRangeException(nameof(value), LogHelper.FormatInvariant(LogMessages.IDX10108, LogHelper.MarkAsNonPII(MinimumAutomaticRefreshInterval), LogHelper.MarkAsNonPII(value))));
 
                 _automaticRefreshInterval = value;
+                Interlocked.Exchange(ref _automaticRefreshIntervalInSeconds, value.TotalSeconds);
             }
         }
 
+        /// <summary>
+        /// Records the time this instance was created.
+        /// Used to determine if the automatic refresh or request refresh interval has passed.
+        /// The set is used for testing purposes.
+        /// </summary>
+        internal DateTimeOffset StartupTime { get; set; } = DateTimeOffset.UtcNow;
+
+        /// <summary>
+        /// Each time GetConfigurationAsync() results in a http request, this value is incremented.
+        /// </summary>
+        internal long NumberOfTimesAutomaticRefreshRequested { get; set; }
```

Comment:
`long` is not guaranteed to be atomic during reads and writes, so if this is being exposed to be consumed externally there may be a threading issue.

Reply:
Telemetry code can use Interlocked.Read

Reply:
Good catch.

Reply:
> Telemetry code can use Interlocked.Read

Sure, but it needs to be documented. Right now, there's no indication of this other than it referencing a method with the `Async` suffix which requires the user to infer that a lock is required on read.

Reply:
good point @msbw2 , unresolving, specifically asking for an update to the doc string.

Reply:
Also, I think the writer (`x++`) needs to lock this on write so that the reader doesn't lock in between write operations.

Reply:
You can do an interlocked increment on the long as well I think to avoid the lock.